instances/widgets.treasury-factory.near/widget/pages/settings/system-updates/DAOContractUpdate.jsx

```diff
@@ -0,0 +1,203 @@
+    }
+  }
+
+  // Add '1' for each leading 0 byte
+  for (let i = 0; i < zeros; i++) {
+    encoded.push(alphabet[0]);
+  }
+
+  // Since we pushed remainders, reverse to get correct order
+  return encoded.reverse().join("");
+}
+
+function getDefaultCodeHash() {
+  return asyncFetch(`${REPL_RPC_URL}`, {
+    method: "POST",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      jsonrpc: "2.0",
+      id: "dontcare",
+      method: "query",
+      params: {
+        request_type: "call_function",
+        finality: "final",
+        account_id: "sputnik-dao.near",
+        method_name: "get_default_code_hash",
+        args_base64: "",
+      },
+    }),
+  }).then((response) => {
+    const dao_code_hash_b58 = JSON.parse(
+      Buffer.from(new Uint8Array(response.body.result.result)).toString()
+    );
+    return dao_code_hash_b58;
+  });
+}
+
+function checkIfDAOContractIsUpToDate(instance) {
+  const {
+    updatesNotApplied,
+    finishedUpdates,
+    proposedUpdates,
+    setFinishedUpdates,
+    setProposedUpdates,
+    UPDATE_TYPE_DAO_CONTRACT,
+  } = VM.require(
+    "${REPL_BASE_DEPLOYMENT_ACCOUNT}/widget/pages.settings.system-updates.UpdateNotificationTracker"
+  ) ?? { updatesNotApplied: [], setFinishedUpdates: () => {} };
+
+  const daoContractUpdatesNotApplied = updatesNotApplied.filter(
+    (update) => update.type === UPDATE_TYPE_DAO_CONTRACT
+  );
+  const { treasuryDaoID } = VM.require(`${instance}/widget/config.data`);
+
+  if (daoContractUpdatesNotApplied.length === 0 || !treasuryDaoID) {
+    return;
+  }
+  getDefaultCodeHash().then((dao_code_hash_b58) => {
+    asyncFetch(`${REPL_RPC_URL}`, {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        jsonrpc: "2.0",
+        id: "dontcare",
+        method: "query",
+        params: {
+          request_type: "view_code",
+          finality: "final",
+          account_id: treasuryDaoID,
+        },
+      }),
+    }).then((response) => {
+      const dao_contract_code_base64 = response.body.result.code_base64;
+      const dao_contract_bytes = Buffer.from(
+        dao_contract_code_base64,
+        "base64"
+      );
+
+      const dao_contract_code_base58 = base58EncodeFromHex(
+        ethers.utils.sha256(dao_contract_bytes)
+      );
+      if (dao_code_hash_b58 === dao_contract_code_base58) {
+        console.log(
+          "The DAO contract hash is identical with the factory default."
+        );
+
+        daoContractUpdatesNotApplied.forEach((update) => {
+          finishedUpdates[update.id] = true;
+        });
+        setFinishedUpdates(finishedUpdates);
+      } else {
+        console.log(
+          "The DAO contract hash is different from the factory default"
+        );
+
+        Near.asyncView(treasuryDaoID, "get_proposals", {
+          from_index: 0,
```

Comment:
can you please add a TODO comment, to update with indexer, so I remember later that I have to update it here as well.